modules/ROOT/pages/subqueries/collect.adoc

```diff
@@ -0,0 +1,318 @@
+[[collect]]
+= COLLECT subqueries
+
+_This feature was introduced in Neo4j 5.6._
+
+A `COLLECT` subquery expression can be used to create a list with the rows returned by a given subquery.
+
+Any non-writing query is allowed.
+`COLLECT` subqueries differ from `COUNT` and `EXISTS` subqueries in that the final `RETURN` clause is mandatory.
+The `RETURN` clause must return exactly one column.
+
+
+[[collect-syntac]]
+
+[source, syntax]
+----
+COLLECT LCURLY regularQuery RCURLY
+----
+
+[[collect-example]]
+== Example graph
+
+The following graph is used for the examples below:
+
+image:subqueries_graph.svg[]
+
+To recreate the graph, run the following query against an empty Neo4j database:
+
+[source, cypher, role=test-setup]
+----
+CREATE
+(andy:Swedish:Person {name: 'Andy', age: 36}),
+(timothy:Person {name: 'Timothy', nickname: 'Tim', age: 25}),
+(peter:Person {name: 'Peter', nickname: 'Pete', age: 35}),
+(andy)-[:HAS_DOG {since: 2016}]->(:Dog {name:'Andy'}),
+(timothy)-[:HAS_CAT {since: 2019}]->(:Cat {name:'Mittens'}),
+(fido:Dog {name:'Fido'})<-[:HAS_DOG {since: 2010}]-(peter)-[:HAS_DOG {since: 2018}]->(:Dog {name:'Ozzy'}),
+(fido)-[:HAS_TOY]->(:Toy{name:'Banana'})
+----
+
+[[collect-simple]]
+== Simple `COLLECT` subquery
+
+Variables introduced by the outside scope can be used in the `COLLECT` subquery without importing them.
+In this regard, `COLLECT` subqueries are different from `CALL` subqueries, xref::subqueries/call-subquery.adoc#call-importing-variables[which do require importing].
+The following query exemplifies this and outputs the owners of the dog named `Ozzy`:
+
+[source, cypher]
+----
+MATCH (person:Person)
+WHERE 'Ozzy' IN COLLECT { MATCH (person)-[:HAS_DOG]->(dog:Dog) RETURN dog.name }
+RETURN person.name AS name
+----
+
+[role="queryresult",options="header,footer",cols="1*<m"]
+|===
+| name
+| "Peter"
+1+d|Rows: 1
+|===
+
+
+[[collect-where]]
+== `COLLECT` subquery with `WHERE` clause
+
+A `WHERE` clause can be used inside the `COLLECT` pattern.
+Variables introduced by the `MATCH` clause and the outside scope can be used in the inner scope.
+
+[source, cypher]
+----
+MATCH (person:Person)
+RETURN person.name as name, COLLECT {
```

Comment:
Noticed that COLLECT doesn't get turned to bold in the query on the generated website, do you know if there are plans to update the parser for that?

Reply:
Great point, the latest update is from 4 years ago :sweat_smile: https://github.com/highlightjs/highlightjs-cypher/blob/master/src/cypher.js
We should definitely bring it up to date. You can for sure spot missing keywords quicker that I can, but otherwise I can have a look at some point :)

Reply:
I don't have much time atm to go through all the keywords, but there is a CIP with all Keywords [here](https://docs.google.com/document/d/1I1OWtBB4hvW-dZ_we7koXka3_zgAVRM3_0Pkg7RTYrM/edit#heading=h.lzdax825k5wi) :) This might help! If more help is needed, my team (Team Cypher Surface) would be the team to reach out to for help